diff --git a/src/gallium/frontends/vdpau/query.c b/src/gallium/frontends/vdpau/query.c
index 6c48e9e8c12..158256b961b 100644
--- a/src/gallium/frontends/vdpau/query.c
+++ b/src/gallium/frontends/vdpau/query.c
@@ -108,8 +108,6 @@ vlVdpVideoSurfaceQueryGetPutBitsYCbCrCapabilities(VdpDevice device, VdpChromaTyp
 {
    vlVdpDevice *dev;
    struct pipe_screen *pscreen;
-   VdpYCbCrFormat ycbcrFormat;
-   bool supported;
 
    if (!is_supported)
       return VDP_STATUS_INVALID_POINTER;
@@ -124,50 +122,47 @@ vlVdpVideoSurfaceQueryGetPutBitsYCbCrCapabilities(VdpDevice device, VdpChromaTyp
 
    mtx_lock(&dev->mutex);
 
-   ycbcrFormat = bits_ycbcr_format;
    switch(bits_ycbcr_format) {
    case VDP_YCBCR_FORMAT_NV12:
-      supported = surface_chroma_type == VDP_CHROMA_TYPE_420;
+      *is_supported = surface_chroma_type == VDP_CHROMA_TYPE_420;
       break;
 
    case VDP_YCBCR_FORMAT_YV12:
-      supported = surface_chroma_type == VDP_CHROMA_TYPE_420;
+      *is_supported = surface_chroma_type == VDP_CHROMA_TYPE_420;
 
       /* We can convert YV12 to NV12 on the fly! */
-      ycbcrFormat = VDP_YCBCR_FORMAT_NV12;
+      if (*is_supported &&
+          pscreen->is_video_format_supported(pscreen,
+                                             PIPE_FORMAT_NV12,
+                                             PIPE_VIDEO_PROFILE_UNKNOWN,
+                                             PIPE_VIDEO_ENTRYPOINT_BITSTREAM)) {
+         mtx_unlock(&dev->mutex);
+         return VDP_STATUS_OK;
+      }
       break;
 
    case VDP_YCBCR_FORMAT_UYVY:
    case VDP_YCBCR_FORMAT_YUYV:
-      supported = surface_chroma_type == VDP_CHROMA_TYPE_422;
+      *is_supported = surface_chroma_type == VDP_CHROMA_TYPE_422;
       break;
 
    case VDP_YCBCR_FORMAT_Y8U8V8A8:
    case VDP_YCBCR_FORMAT_V8U8Y8A8:
-      supported = surface_chroma_type == VDP_CHROMA_TYPE_444;
-      break;
-
-   case VDP_YCBCR_FORMAT_P010:
-   case VDP_YCBCR_FORMAT_P016:
-      /* Do any other profiles imply support for this chroma type? */
-      supported = (surface_chroma_type == VDP_CHROMA_TYPE_420_16)
-                  && vl_codec_supported(pscreen, PIPE_VIDEO_PROFILE_HEVC_MAIN_10, false);
+      *is_supported = surface_chroma_type == VDP_CHROMA_TYPE_444;
       break;
 
    default:
-      supported = false;
+      *is_supported = false;
       break;
    }
 
-   if (supported &&
+   if (*is_supported &&
        !pscreen->is_video_format_supported(pscreen,
-                                           FormatYCBCRToPipe(ycbcrFormat),
+                                           FormatYCBCRToPipe(bits_ycbcr_format),
                                            PIPE_VIDEO_PROFILE_UNKNOWN,
                                            PIPE_VIDEO_ENTRYPOINT_BITSTREAM)) {
-      supported = false;
+      *is_supported = false;
    }
-   *is_supported = supported;
-
    mtx_unlock(&dev->mutex);
 
    return VDP_STATUS_OK;
