diff --git a/Engine/Build/BatchFiles/Linux/Setup.sh b/Engine/Build/BatchFiles/Linux/Setup.sh
index 18dc22fa5019..f8461024d716 100755
--- a/Engine/Build/BatchFiles/Linux/Setup.sh
+++ b/Engine/Build/BatchFiles/Linux/Setup.sh
@@ -41,7 +41,7 @@ if [ -e /etc/os-release ]; then
 fi
 
 # Install our bundled toolchain unless we are running a Perforce build or a an our of source toolchain is available.
-if [ ! -f Build/PerforceBuild.txt ] && [ -z "$LINUX_MULTIARCH_ROOT" ] && [ -z "$UE_SDKS_ROOT" ]; then
+if [ ! -f Build/PerforceBuild.txt ] && [[ "$ID" != "arch" ]] && [ -z "$LINUX_MULTIARCH_ROOT" ] && [ -z "$UE_SDKS_ROOT" ]; then
   echo "Installing a bundled clang toolchain"
   pushd Build/BatchFiles/Linux > /dev/null
   ./SetupToolchain.sh
