diff --git a/Setup.sh b/Setup.sh
index f91a96aaac6b..37df72d1337b 100755
--- a/Setup.sh
+++ b/Setup.sh
@@ -80,6 +80,7 @@ else
        echo Register the engine installation...
        if [ -f Engine/Binaries/Linux/UnrealVersionSelector-Linux-Shipping ]; then
                pushd Engine/Binaries/Linux > /dev/null
+               make
                ./UnrealVersionSelector-Linux-Shipping -register > /dev/null &
                popd > /dev/null
        fi
diff --git a/Engine/Source/Programs/UnrealBuildTool/Platform/Linux/LinuxToolChain.cs b/Engine/Source/Programs/UnrealBuildTool/Platform/Linux/LinuxToolChain.cs
index 1b19b5895ccc..e1c27631606d 100644
--- a/Engine/Source/Programs/UnrealBuildTool/Platform/Linux/LinuxToolChain.cs
+++ b/Engine/Source/Programs/UnrealBuildTool/Platform/Linux/LinuxToolChain.cs
@@ -58,7 +58,7 @@ namespace UnrealBuildTool
                        if (CompilerVersionLessThan(15, 0, 0) || CompilerVersionGreaterOrEqual(17, 0, 6))
                        {
                                throw new BuildException(
-                                       String.Format("This version of the Unreal Engine can only be compiled with clang 15.x or 16.x. clang {0} may not build it - please use>
+                                       String.Format("This version of the Unreal Engine can only be compiled with clang 15.x or 17.x. clang {0} may not build it - please use>
                                                Info.ClangVersion)
                                        );
                        }
@@ -819,7 +819,7 @@ namespace UnrealBuildTool
                                }
                                else
                                {
-                                       Arguments.Add("-Wl,-no-pie");
+                                       Arguments.Add("-Wl,--no-pie");
                                }
                        }
diff --git a/Engine/Source/Runtime/Core/Public/HAL/Platform.h b/Engine/Source/Runtime/Core/Public/HAL/Platform.h
index ee3e2ea06c5d..ac2a1724eba6 100644
--- a/Engine/Source/Runtime/Core/Public/HAL/Platform.h
+++ b/Engine/Source/Runtime/Core/Public/HAL/Platform.h
@@ -1069,7 +1069,7 @@ namespace TypeTests
 #endif
        static_assert((!TAreTypesEqual_V<ANSICHAR, UCS2CHAR>),  "ANSICHAR and UCS2CHAR should be different types.");
        static_assert((!TAreTypesEqual_V<WIDECHAR, UCS2CHAR>),  "WIDECHAR and UCS2CHAR should be different types.");
-       static_assert(TAreTypesEqual_V<TCHAR, ANSICHAR> || TAreTypesEqual_V<TCHAR, WIDECHAR> || TAreTypesEqual_V<TCHAR, UTF8CHAR>, "TCHAR should either be ANSICHAR, WIDECHAR >
+       static_assert(TAreTypesEqual_V<TCHAR, ANSICHAR> | TAreTypesEqual_V<TCHAR, WIDECHAR> | TAreTypesEqual_V<TCHAR, UTF8CHAR>, "TCHAR should either be ANSICHAR, WIDECHAR or>
 
 #if PLATFORM_WIDECHAR_IS_CHAR16
        static_assert(TAreTypesEqual_V<WIDECHAR, char16_t>, "WIDECHAR should be char16_t");

