diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4322e4b6..10711e50 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -150,8 +150,8 @@ if(RESOURCE_INSTALL_DIR)
 	install(DIRECTORY shaders/ DESTINATION ${RESOURCE_INSTALL_DIR}/shaders/)
 else()
 	if(NOT USE_RELATIVE_ASSET_PATH)
-		add_definitions(-DVK_EXAMPLE_ASSETS_DIR=\"${CMAKE_SOURCE_DIR}/assets/\")
-		add_definitions(-DVK_EXAMPLE_SHADERS_DIR=\"${CMAKE_SOURCE_DIR}/shaders/\")
+		add_definitions(-DVK_EXAMPLE_ASSETS_DIR=\"/usr/share/vulkan/data/assets/\")
+		add_definitions(-DVK_EXAMPLE_SHADERS_DIR=\"/usr/share/vulkan/data/shaders/\")
 	endif()
 endif()
 
